lib/spack/docs/packaging_guide.rst

```diff
@@ -880,6 +882,12 @@ Git fetching supports the following parameters to ``version``:
 * ``tag``: Name of a tag to fetch.
 * ``commit``: SHA hash (or prefix) of a commit to fetch.
 * ``submodules``: Also fetch submodules recursively when checking out this repository.
+* ``full_depth``: Ensure the full git history is checked out. Normally
```

Comment:
Are you seeing cases where you only want to enable one of `full_depth` and `all_branches`? If not, IMO it would make sense to combine these into one option like `get_full_repo`.

Reply:
Not so far. I've combined them. I extended the test to multiple versions of `git` though (and extended the `git_version` fixture to return the currently-selected version), because the behavior is not obvious for `1.7.1 <= git_version < 1.7.10`.